cmd/manager/main.go

```diff
@@ -186,6 +192,30 @@ func main() {
 	}
 }
 
+func addMonitoringKeyLabelToOperatorService(cfg *rest.Config, service *v1.Service) error {
+	if service == nil {
+		return fmt.Errorf("service doesn't exist")
+	}
+
+	kclient, err := client.New(cfg, client.Options{})
+	if err != nil {
+		return err
+	}
+
+	updatedLabels := map[string]string{"monitoring-key": "middleware"}
+	for k, v := range service.ObjectMeta.Labels {
+		updatedLabels[k] = v
+	}
+	service.ObjectMeta.Labels = updatedLabels
+
+	err = kclient.Update(context.TODO(), service)
```

Comment:
In the main function we already assign `context.TODO()` to a variable `ctx`. Could you please reuse it here?

Reply:
good catch, fixed now